ntbs-service/Pages/Search/Index.cshtml

```diff
@@ -13,60 +14,64 @@
-            </th>
-        </tr>
-    </thead>
-    <tbody>
-@foreach (var item in Model.Results) {
-        <tr>
-            <td>
-                @Html.DisplayFor(modelItem => item.Source)
-            </td>
-            <td>
-                @Html.DisplayFor(modelItem => item.FamilyNames)
-            </td>
-            <td>
-                @Html.DisplayFor(modelItem => item.GivenNames)
-            </td>
-            <td>
-                @Html.DisplayFor(modelItem => item.NhsNumber)
-            </td>
-            <td>
-                @Html.DisplayFor(modelItem => item.DateOfBirth)
-            </td>
-            <td>
-                @Html.DisplayFor(modelItem => item.DateOfNotification)
-            </td>
-        </tr>
+<div>
+    <div class="clear-search-button-container">
+        <a href="/search" class="clear-search-button">
+            Clear search & start again
+        </a>
+    </div>
+    <div>
+        <p> @Model.SearchResultsBannerDisplay.NumberOfResults Results found. </p>
+        <p> To filter results, add more search terms and search again. </p>
+    </div>
+    <ol class="search-results-list">
+        @foreach(var result in @Model.SearchResultsBannerDisplay) {
+            <li>
+                <partial name="../Shared/_NotificationBanner.cshtml" model="result"/>
+            </li>
+        }
+    <ol>
+
+    <nav class="nhsuk-pagination" role="navigation" aria-label="Pagination">
+        <ul class="nhsuk-list nhsuk-pagination__list">
+            @{
+
+                var previousPageText = "";
+                var previousPageUrl = "";
+                var nextPageText = "";
+                var nextPageUrl = "";
+                var queryString = Request.Query;
+                var previousPageQueryString = new RouteValueDictionary();
+                queryString.Keys.Select(x => previousPageQueryString[x] = queryString[x].ToString());
+                var nextPageQueryString = previousPageQueryString;
+                if(@Model.SearchResultsBannerDisplay.HasPreviousPage)
+                {
+                    previousPageText = "Page " + (Model.SearchResultsBannerDisplay.PageIndex - 1) + " of " + (Model.SearchResultsBannerDisplay.TotalPages);
+                    previousPageQueryString["pageIndex"] = "" + (Model.SearchResultsBannerDisplay.PageIndex - 1);
+                    previousPageUrl = @Url.Action("Search", previousPageQueryString);
+                }
+                if(@Model.SearchResultsBannerDisplay.HasNextPage)
+                {
+                    nextPageText = "Page " + (Model.SearchResultsBannerDisplay.PageIndex + 1) + " of " + (Model.SearchResultsBannerDisplay.TotalPages);
+                    nextPageQueryString["pageIndex"] = "" + (Model.SearchResultsBannerDisplay.PageIndex + 1);
+                    nextPageUrl = @Url.Action("Search", nextPageQueryString);
+                }
+            }
+            <nhs-pagination next-link-text="@nextPageText" next-url="@nextPageUrl" 

```

Comment:
passing empty strings into next-link-text/next-url causes the next page pagination to not appear which suits us. The general format of this block of code here is quite grim - happy for any input on refactoring or bringing this query string construction to happen in the cshtml.cs file in the onGet method and adding it to the model?